src/controllers/legacycontrollermappingfilehandler.cpp

```diff
@@ -100,6 +101,85 @@ void LegacyControllerMappingFileHandler::parseMappingInfo(
     mapping->setWikiLink(wiki.isNull() ? "" : wiki.text());
 }
 
+void LegacyControllerMappingFileHandler::parseMappingSettings(
+        const QDomElement& root, LegacyControllerMapping* mapping) const {
+    if (root.isNull() || !mapping) {
+        return;
+    }
+
+    QDomElement settings = root.firstChildElement("settings");
+    if (settings.isNull()) {
+        return;
+    }
+
+    std::unique_ptr<LegacyControllerSettingsLayoutContainer> settingLayout =
+            std::make_unique<LegacyControllerSettingsLayoutContainer>(
+                    LegacyControllerSettingsLayoutContainer::Disposition::
+                            VERTICAL);
+    parseMappingSettingsElement(settings, mapping, settingLayout.get());
+    mapping->setSettingLayout(std::move(settingLayout));
+}
+
+void LegacyControllerMappingFileHandler::parseMappingSettingsElement(
+        const QDomElement& current,
+        LegacyControllerMapping* pMapping,
+        LegacyControllerSettingsLayoutContainer* pLayout)
+        const {
+    for (QDomElement element = current.firstChildElement();
+            !element.isNull();
+            element = element.nextSiblingElement()) {
+        const QString& tagName = element.tagName().toLower();
+        if (tagName == "option") {
+            std::shared_ptr<AbstractLegacyControllerSetting> pSetting(
```

Comment:
This can be a std::unique_ptr and later moved into a vector. QList does not support move.